### PR TITLE
bus: Change IBus unique name to :1.0 from IBUS_SERVICE_IBUS

### PR DESCRIPTION
**Pull request by fujiwarat**
According to the D-Bus Specification,
`org.freedesktop.DBus.GetNameOwner()` must return a unique name (i.e. one of the form `:1.23`). The only unique name which is allowed to not be in this form is `org.freedesktop.DBus`.

GLib2 2.80.1 has updated g_dbus_is_unique_name() to meet the sepcification.

In the long term, it might be more robust to rework the code so that the mapping from `org.freedesktop.IBus` → a unique name is stored in `dbus->names` (in `dbusimpl.c`) like all the other requested names. However, handling for the `org.freedesktop.IBus` well-known name is hardcoded throughout `dbusimpl.c`, so porting this single bit of it to create a `BusNameService` would probably cause more problems.

BUG=https://github.com/ibus/ibus/issues/2639
BUG=https://gitlab.gnome.org/GNOME/glib/-/issues/3268 BUG=https://github.com/advisories/GHSA-f632-c3rh-r2v2

### CODE REVIEW

**Comment by pwithnall** [comment]
From a quick look through, this looks good to me. The suggestion on #2639 was to use `:0.0` rather than `:1.0` as `IBUS_NAME_OWNER_NAME`, but `:1.0` should work if you can guarantee that the IBus bus implementation will never allocate that unique name to a client.

File: bus/dbusimpl.c [hunk 1]
```diff
@@ -710,7 +710,7 @@ bus_dbus_impl_list_names (BusDBusImpl           *dbus,
 
     /* FIXME should add them? */
     g_variant_builder_add (&builder, "s", "org.freedesktop.DBus");
-    g_variant_builder_add (&builder, "s", "org.freedesktop.IBus");
+    g_variant_builder_add (&builder, "s", IBUS_NAME_OWNER_NAME);
```
[the remark this answers is not in the document]
Comment (fujiwarat) [reply bus/dbusimpl.c:713]: Thank you for the suggestion.

**Comment by fujiwarat** [comment]
@pwithnall @smcv Thank you for the reviews.

I pushed the candidate patch to Fedora 40 as ibus-1.5.30-2.fc40 for the test purpose.
Probably I will integrate the patch to the upstream one or two weeks later after we don't see any regression issues.

**Comment by fujiwarat** [comment]
https://github.com/ibus/ibus/commit/927ce008e62c07c9f4928be441ce57d0d946ab6d
https://github.com/ibus/ibus/commit/2b146251a37dc8e590270f16a7f7c0be750c63d9
https://github.com/ibus/ibus/commit/7f44459f53ffbc94c300db96d34f5c02afd33b22

**Pull request closed without merging**